django/db/migrations/serializer.py

```diff
@@ -154,7 +160,9 @@ def serialize(self):
                 "import %s" % module
             }
         # Further error checking
-        if self.value.__name__ == "<lambda>":
+        # use .co_name, since this can not be modified, whereas __name__ can be altered
+        # https://stackoverflow.com/a/56451124/67579
+        if self.value.__code__.co_name == "<lambda>":
```

Comment:
This is not related with `EnumSerializer` and cannot be proceeded in the same PR. Also, assigning lambdas to a variable and changing their `__name__` sounds really niche. I'm not sure if it's worth changing :thinking: and we use the same logic in `django.contrib.admin`.

Nevertheless, all bugfixes require [an accepted ticket](https://code.djangoproject.com/newticket), please create a new one.

Reply:
```suggestion
        if self.value.__name__ == "<lambda>":
```